src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/transpose_kernel.hpp

```diff
@@ -0,0 +1,254 @@
+    r7 = _mm512_shuffle_i32x4(t7, tf, 0x88); //   7 ...
+    r8 = _mm512_shuffle_i32x4(t0, t8, 0xdd); //   8 ...
+    r9 = _mm512_shuffle_i32x4(t1, t9, 0xdd); //   9 ...
+    ra = _mm512_shuffle_i32x4(t2, ta, 0xdd); //  10 ...
+    rb = _mm512_shuffle_i32x4(t3, tb, 0xdd); //  11 ...
+    rc = _mm512_shuffle_i32x4(t4, tc, 0xdd); //  12 ...
+    rd = _mm512_shuffle_i32x4(t5, td, 0xdd); //  13 ...
+    re = _mm512_shuffle_i32x4(t6, te, 0xdd); //  14 ...
+    rf = _mm512_shuffle_i32x4(t7, tf, 0xdd); //  15  31  47  63  79  96 111 127 ... 255
+}
+
+template<typename T>
+inline void transpose_16x16_kernel(float* _dst, T* src, size_t dst_stride, size_t src_stride) {
+    auto* dst = reinterpret_cast<uint32_t*>(_dst);
+    __m512i r0, r1, r2, r3, r4, r5, r6, r7, r8, r9, ra, rb, rc, rd, re, rf;
+    r0 = _mm512_castps_si512(mm512_uni_loadu_ps(src));
+    r1 = _mm512_castps_si512(mm512_uni_loadu_ps(src + src_stride));
+    r2 = _mm512_castps_si512(mm512_uni_loadu_ps(src + 2 * src_stride));
+    r3 = _mm512_castps_si512(mm512_uni_loadu_ps(src + 3 * src_stride));
+    r4 = _mm512_castps_si512(mm512_uni_loadu_ps(src + 4 * src_stride));
+    r5 = _mm512_castps_si512(mm512_uni_loadu_ps(src + 5 * src_stride));
+    r6 = _mm512_castps_si512(mm512_uni_loadu_ps(src + 6 * src_stride));
+    r7 = _mm512_castps_si512(mm512_uni_loadu_ps(src + 7 * src_stride));
+    r8 = _mm512_castps_si512(mm512_uni_loadu_ps(src + 8 * src_stride));
+    r9 = _mm512_castps_si512(mm512_uni_loadu_ps(src + 9 * src_stride));
+    ra = _mm512_castps_si512(mm512_uni_loadu_ps(src + 10 * src_stride));
+    rb = _mm512_castps_si512(mm512_uni_loadu_ps(src + 11 * src_stride));
+    rc = _mm512_castps_si512(mm512_uni_loadu_ps(src + 12 * src_stride));
+    rd = _mm512_castps_si512(mm512_uni_loadu_ps(src + 13 * src_stride));
+    re = _mm512_castps_si512(mm512_uni_loadu_ps(src + 14 * src_stride));
+    rf = _mm512_castps_si512(mm512_uni_loadu_ps(src + 15 * src_stride));
+
+    transpose_m512i_16x16(r0, r1, r2, r3, r4, r5, r6, r7, r8, r9, ra, rb, rc, rd, re, rf);
+
+    _mm512_storeu_si512(dst, r0);
+    _mm512_storeu_si512(dst + dst_stride, r1);
+    _mm512_storeu_si512(dst + 2 * dst_stride, r2);
+    _mm512_storeu_si512(dst + 3 * dst_stride, r3);
+    _mm512_storeu_si512(dst + 4 * dst_stride, r4);
+    _mm512_storeu_si512(dst + 5 * dst_stride, r5);
+    _mm512_storeu_si512(dst + 6 * dst_stride, r6);
+    _mm512_storeu_si512(dst + 7 * dst_stride, r7);
+    _mm512_storeu_si512(dst + 8 * dst_stride, r8);
+    _mm512_storeu_si512(dst + 9 * dst_stride, r9);
+    _mm512_storeu_si512(dst + 10 * dst_stride, ra);
+    _mm512_storeu_si512(dst + 11 * dst_stride, rb);
+    _mm512_storeu_si512(dst + 12 * dst_stride, rc);
+    _mm512_storeu_si512(dst + 13 * dst_stride, rd);
+    _mm512_storeu_si512(dst + 14 * dst_stride, re);
+    _mm512_storeu_si512(dst + 15 * dst_stride, rf);
+}
+
+inline void transpose_16x16_kernel(uint32_t* dst, uint32_t* src, size_t dst_stride, size_t src_stride) {
```

Comment:
What will happen if input data is not aligned by 16 * 16 ? Maybe a comment should be added if this function requires memory aligned

Reply:
We checked in  block_size must be multiple of 32 and head_size must be multiple of 16 in [AttentionExecutor::execute ](https://github.com/openvinotoolkit/openvino/blob/c23e096b4289bfadb04435d19eca94894135a86a/src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/executor_pa.cpp#L1508-L1509) which should satisfy the [llm model needs](https://github.com/ilya-lavrenov/vllm/blob/2e5648ab25a0e889f4212ef74530f01ebf638dcd/vllm/model_executor/layers/attention/ops/paged_attn.py#L15), so we can safely use 16x16 kernels.